core/sql/optimizer/NATable.cpp

```diff
@@ -3676,6 +3680,76 @@ NAType* getSQColTypeForHive(const char* hiveType, NAMemory* heap)
                                    CharInfo::IMPLICIT);
   } 
 
+  if ( !strncmp(hiveType, "decimal", 7) )
+  {
+    Lng32 i=0, pstart=0, pend=0, sstart=0, send=0, p=-1, s = -1;
+    Lng32 hiveTypeLen = strlen(hiveType);
+    char pstr[MAX_NUM_LEN], sstr[MAX_NUM_LEN];
+    memset(pstr,0,sizeof(pstr));
+    memset(sstr,0,sizeof(sstr));
+
+    for( i = 0; i < hiveTypeLen; i++ )
+    {
+      if(hiveType[i] == '(' )
+      {
+        pstart = i+1;
+      }
+      else if(hiveType[i] == ',')
+      {
+        pend = i;
+        sstart = i+1;
+      }
+      else if(hiveType[i] == ')')
+      {
+        send = i;
+      }
+      else
+       continue;
+    }
+
+    if(pend - pstart > 0)
```

Comment:
@zellerh , I found Hive will do a kind of normalization here. If we use DECIMAL(4) in DDL, hive will transform it into decimal(4,0) and that is what Trafodion will get and parse it.
So all the checkings here are not necessary, but I still want to leave it since it doesn't harm and not impact performance  too much, this is not hot path.